src/ConnectionLoader.js

```diff
@@ -5,6 +5,24 @@ const FIRST = '$$FIRST$$';
 
 export default class ConnectionLoader {
   constructor(singleConnectionHandler) {
+    const fetch = async ({ connection, options, ...props }) => {
+      const result = connection
+        ? await singleConnectionHandler({ connection, options, ...props })
+        : await singleConnectionHandler({ options, ...props });
+
+      if (result.continuationToken && !result.items.length) {
```

Comment:
The returned items are not always in a property named "items".  [Bug 1436212](https://bugzilla.mozilla.org/show_bug.cgi?id=1436212) has some background on why this is hard.

Reply:
Every connection loader in `taskcluster-web-server` returns an items property (https://github.com/taskcluster/taskcluster-web-server/search?utf8=%E2%9C%93&q=items%3A&type=). I guess we can write tests to assert that each connection handler returns an object with an `items` fields.

Reply:
Oh, awesome.  I didn't know these were results from a loader.  One of the downsides of this heavily-functional style of programming is that it's not easy to follow control flow!